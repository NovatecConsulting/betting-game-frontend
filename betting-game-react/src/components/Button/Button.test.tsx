import React from "react";
import { shallow } from "enzyme";
import { Button } from ".";

describe(Button.name + " component", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });
    it("should render and match snapshot with label and onClick method.", () => {
        const onClick = jest.fn();

        const wrapper = shallow(<Button label="Test" onClick={onClick} />);

        wrapper.find("div").simulate("click");

        expect(onClick).toHaveBeenCalled();
        expect(wrapper.find("div")).toIncludeText("Test");
        expect(wrapper).toMatchSnapshot();
    });
});
