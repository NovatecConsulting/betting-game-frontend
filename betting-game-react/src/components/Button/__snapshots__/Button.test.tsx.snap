// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render and match snapshot with label and onClick method. 1`] = `
<div
  className="cursor-pointer inline-block text-sm px-5 py-2 leading-none border rounded-full text-white border-white hover:text-blue-400 hover:bg-white active:bg-blue-700 "
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Test
</div>
`;
